op-challenger/cmd/move.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ethereum-optimism/optimism/op-challenger/flags"
+	"github.com/ethereum-optimism/optimism/op-challenger/game/fault/contracts"
+	opservice "github.com/ethereum-optimism/optimism/op-service"
+	"github.com/ethereum-optimism/optimism/op-service/dial"
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	"github.com/ethereum-optimism/optimism/op-service/sources/batching"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr"
+	"github.com/ethereum-optimism/optimism/op-service/txmgr/metrics"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/urfave/cli/v2"
+)
+
+var (
+	AttackFlag = &cli.BoolFlag{
+		Name:    "attack",
+		Usage:   "An attack move. If true, the defend flag must not be set.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "ATTACK"),
+	}
+	DefendFlag = &cli.BoolFlag{
+		Name:    "defend",
+		Usage:   "A defending move. If true, the attack flag must not be set.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "DEFEND"),
+	}
+	ParentIndexFlag = &cli.StringFlag{
+		Name:    "parent-index",
+		Usage:   "The index of the claim to move on.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "PARENT_INDEX"),
+	}
+	ClaimFlag = &cli.StringFlag{
+		Name:    "claim",
+		Usage:   "The claim hash.",
+		EnvVars: opservice.PrefixEnvVar(envVarPrefix, "CLAIM"),
+	}
+)
+
+func Move(ctx *cli.Context) error {
+	logger, err := setupLogging(ctx)
+	if err != nil {
+		return err
+	}
+	rpcUrl := ctx.String(flags.L1EthRpcFlag.Name)
+	if rpcUrl == "" {
+		return fmt.Errorf("missing %v", flags.L1EthRpcFlag.Name)
+	}
+	gameAddr, err := opservice.ParseAddress(ctx.String(GameAddressFlag.Name))
+	if err != nil {
+		return err
+	}
+
+	attack := ctx.Bool(AttackFlag.Name)
+	defend := ctx.Bool(DefendFlag.Name)
+	parentIndex := ctx.Uint64(ParentIndexFlag.Name)
+	claim := common.HexToHash(ctx.String(ClaimFlag.Name))
+
+	if attack && defend {
+		return fmt.Errorf("both attack and defense flags cannot be set")
+	}
+
+	l1Client, err := dial.DialEthClientWithTimeout(ctx.Context, dial.DefaultDialTimeout, logger, rpcUrl)
+	if err != nil {
+		return fmt.Errorf("failed to dial L1: %w", err)
+	}
+	defer l1Client.Close()
+
+	caller := batching.NewMultiCaller(l1Client.Client(), batching.DefaultBatchSize)
+	contract, err := contracts.NewFaultDisputeGameContract(gameAddr, caller)
+	if err != nil {
+		return fmt.Errorf("failed to create dispute game bindings: %w", err)
+	}
+
+	txMgrConfig := txmgr.ReadCLIConfig(ctx)
+	txMgr, err := txmgr.NewSimpleTxManager("challenger", logger, &metrics.NoopTxMetrics{}, txMgrConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create the transaction manager: %w", err)
+	}
```

Comment:
It's really tempting to see if we can extract all this common setup logic into a shared method. Even if it's only shared between these subcommands.

Reply:
Done, I think it's now significantly cleaner. Thanks for pushing me to do this 😄